protected/humhub/libs/StatableTrait.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) 2023 HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\libs;
+
+use humhub\components\ActiveRecord;
+use humhub\interfaces\StatableActiveQueryInterface;
+use humhub\interfaces\StateServiceInterface;
+use yii\base\InvalidConfigException;
+use yii\db\ActiveQuery;
+use yii\validators\InlineValidator;
+
+/**
+ * @property int $state
+ * @since 1.16
+ */
+trait StatableTrait
+{
+    /**
+     * @inheritdoc
+     */
+    public function canChangeState($state, array $options = []): bool
+    {
+        return $state !== null && $state !== '' && $this->getStateService()->canChange($state, $options);
+    }
+
+    public function getStateService(): StateServiceInterface
+    {
+        return static::getStateServiceClass()::factory($this);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validateStateAttribute(string $attribute, $params, InlineValidator $validator, $current): bool
```

Comment:
Is this used?

Reply:
Yes, it's used every time a whereStatus/andWhereStatus (and hence checkStatus) is called. It is also available to be used as a validator in the rules() section. This should indeed be added in the changed Models:

- [ ] User
- [ ] Space
- [ ] Membership
- [ ] Content